homeassistant/components/roborock/sensor.py

```diff
@@ -174,7 +174,8 @@ class RoborockSensorDescription(
         icon="mdi:garage-open",
         translation_key="dock_error",
         value_fn=lambda data: data.status.dock_error_status.name
-        if data.status.dock_type != RoborockDockTypeCode.no_dock
+        if data.status.dock_error_status is not None
```

Comment:
We typically don't want ternary lambdas. Can we make this into a function?

Reply:
Done. Can you review it again?